docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,411 @@
+
+Since we're on the topic, you must use a similar strategy in your `App.js` file to conditionally render a DOM element to let the user know they are in a preview state. This must happen once per instance of the App, since we will not perpetuate the `cms-preview` query parameter on route changes.
+
+Class component:
+
+```js
+// App.js
+class App extends Nextapp {
+    state = {
+        isPreviewingContentful = false,
+    };
+
+    // Using the componentWillMount() lifecycle function guarantees that this runs only once per instance of App
+    componentWillMount() {
+        const { router } = this.props;
+
+        this.setState({
+            isPreviewingContentful: Object.hasOwnProperty.call(router.query, 'cms-preview');
+        });
+    }
+
+    render() {
+        return (
+            (...)
+                { this.isPreviewingContentful && <ContentfulPreview> }
+            (...)
+        );
+    }
+}
+```
+
+Using hooks: 
+
+```js
+// App.js
+const App = ({ Component, pageProps, rootSelector, router }) => {
+    const [isPreviewingContentful, setIsPreviewingContentful] = useState(false);
+    
+    // Using a useEffect hook with no dependencies guaranties that it fires only once per instance of App
+    useEffect(() => setIsPreviewingContentful(Object.hasOwnProperty.call(router.query, 'cms-preview')), [])
+
+    return (
+            (...)
+                { this.isPreviewingContentful && <ContentfulPreview> }
+            (...)
+        )
+}
+```
+
+### 4. CMS Translations
+
+This boilerplate already includes **Internationalization** using [`@moxy/next-intl`](https://github.com/moxystudio/next-intl), this is done by staticaly configuring individual `intl/messages/<locale>.json` per locale. You can do this dynamically by moving this static files to a **Contentful** content model and by enabling localization in it.
+
+#### 1. Defining the content type
+
+Firstly define the content type that will consist on a list of key/value pairs. Add the fields necessary to fit your needs, for example `homePageTitle`.
+
+> ℹ️ Make sure that you enable localization on every field you set in your content type.
+
+After creating your content type populate the respective fields accordingly.
+
+#### 2. Overriding [`@moxy/next-intl`](https://github.com/moxystudio/next-intl)
+
+Now that you have your content type set you should update the `/intl/index.js` file to fetch the data from **Contentful**. This is done by using the [**Contentful** SDK](https://www.npmjs.com/package/contentful).
+
+```js
+locales: [
+    {
+        id: 'en-US',
+        name: 'English',
+        loadMessages: async () => {
+            const content = await client.getEntries({
+                content_type: 'genericTranslations',
+                locale: 'en-US',
+                // The parameter bellow is needed to avoid the problem described at the top of the file.
+            });
+
+            return content.items[0].fields; melhorar isto
```

Comment:
Do you still need to work on this?

Reply:
actually these are my leftovers :D @dominguesgm i'm sorry i didn't remove this from here on the handoff, please remove now 🙏